src/AppInstallerCLICore/Workflows/DownloadFlow.cpp

```diff
@@ -268,6 +257,25 @@ namespace AppInstaller::CLI::Workflow
         }
     }
 
+    void DownloadInstaller(Execution::Context& context)
+    {
+        // Check if file was already downloaded.
+        // This may happen after a failed installation or if the download was done
+        // separately before, e.g. on COM scenarios.
+        context <<
+            ReportExecutionStage(ExecutionStage::Download) <<
+            CheckForExistingInstaller;
+
+        if (context.IsTerminated())
+        {
+            return;
+        }
+
+        context <<
+            ExecuteUriValidation(UriValidationSource::PackageCatalogSource) <<
+            DownloadInstallerInternal;

```

Comment:
`<<` only executes if the context isn't terminated
```suggestion
        context <<
            ReportExecutionStage(ExecutionStage::Download) <<
            CheckForExistingInstaller <<
            ExecuteUriValidation(UriValidationSource::PackageCatalogSource) <<
            DownloadInstallerInternal;
```